src/data/fin_enum.lean

```diff
@@ -0,0 +1,205 @@
+/-
+Copyright (c) 2019 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Simon Hudon
+-/
+import category.monad.basic
+import data.list.basic
+import data.equiv.basic
+import data.finset
+import data.fintype
+
+/-!
+Type class for finitely enumerable types. The property is stronger
+than `fintype` in that it assigns each element a rank in a finite
+enumeration.
+-/
+
+open finset (hiding singleton)
+
+/-- `fintype α` means that `α` is finite and can be enumerate in some order,
+  i.e. `α` can be mapped onto a finite interval of `ℕ`. -/
```

Comment:
"mapped onto a finite interval of N" means something quite different than what you're actually saying here. Why not just say "i.e. `α` has an explicit bijection with `fin n` for some `n`"?